src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -85,7 +85,8 @@ internal sealed partial class ServiceHubRemoteHostClient : RemoteHostClient
             // to UI thread to run. 
             await Task.Factory.SafeStartNew(() =>
             {
-                vsWorkspace.GetProjectTrackerAndInitializeIfNecessary(Shell.ServiceProvider.GlobalProvider).RegisterWorkspaceHost(host);
+                var projectTracker = vsWorkspace.GetProjectTrackerAndInitializeIfNecessary(Shell.ServiceProvider.GlobalProvider);
+                projectTracker.RegisterWorkspaceHost(host, readyForEvents: true);
```

Comment:
Why add a new method vs. just immediately calling `projectTracker.StartSendingEventsToWorkspaceHost(host)`? My worry is by doing this asynchronously you're skipping the "catch the host back up" logic in the StartSendingEventsToWorkspaceHost method, which might screw up our state, even if you don't care.